cns/restserver/internalapi.go

```diff
@@ -195,28 +196,43 @@ func (service *HTTPRestService) syncHostNCVersion(ctx context.Context, channelMo
 	for _, nc := range ncList.Containers {
 		newHostNCVersionList[nc.NetworkContainerID] = nc.Version
 	}
-	for _, ncID := range hostVersionNeedsUpdateContainers {
+	for ncID := range hostVersionNeedsUpdateContainers {
 		versionStr, ok := newHostNCVersionList[ncID]
 		if !ok {
+			// NMA doesn't have this NC that we need programmed yet, bail out
 			continue
 		}
 		version, err := strconv.Atoi(versionStr)
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse container version of %s", ncID)
 		}
-
 		// Check whether it exist in service state and get the related nc info
 		ncInfo, exist := service.state.ContainerStatus[ncID]
 		if !exist {
 			return errors.Wrapf(errNonExistentContainerStatus, "can't find NC with ID %s in service state, stop updating this host NC version", ncID)
 		}
+		hostVersion, err := strconv.Atoi(ncInfo.HostVersion)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse host nc version string %s", ncInfo.HostVersion)
+		}
+		if hostVersion > version {
+			logger.Errorf("NC version from NMA is decreasing: have %d, got %d", hostVersion, version)
```

Comment:
is hostVersion the last good version provided by host or it is same as nc?

Reply:
If former, then instead of comparing it with hostVersion we need to compare it with NC version to match it is same or higher than NC new version

Reply:
I think this logic is already covered from line 180 to 182

Reply:
> If former, then instead of comparing it with hostVersion we need to compare it with NC version to match it is same or higher than NC new version

I think this logic is already covered from line 180 to 182

Reply:
Compare the hostVersion with version is to check whether NMA's version gone backwards. For example, last NMA's version is 10 and we already update all the pending programming IPs, then we got a version 11 from DNC, then CNS try to sync NMA, NMA returns version 9 and we want to  log this error.

Reply:
> If former, then instead of comparing it with hostVersion we need to compare it with NC version to match it is same or higher than NC new version
Compare the hostVersion with version is to check whether NMA's version gone backwards. For example, last NMA's version is 10 and we already update all the pending programming IPs, then we got a version 11 from DNC, then CNS try to sync NMA, NMA returns version 9 and we want to log this error.

Reply:
i got the flow, this looks good